cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/co/cask/cdap/datastreams/SparkStreamingPipelineDriver.java

```diff
@@ -92,13 +94,17 @@ public void run(final JavaSparkExecutionContext sec) throws Exception {
 
       // there isn't any way to instantiate the fileset except in a TxRunnable, so need to use a reference.
       final AtomicReference<Location> checkpointBaseRef = new AtomicReference<>();
-      sec.execute(new TxRunnable() {
-        @Override
-        public void run(DatasetContext context) throws Exception {
-          FileSet checkpointFileSet = context.getDataset(DataStreamsApp.CHECKPOINT_FILESET);
-          checkpointBaseRef.set(checkpointFileSet.getBaseLocation());
-        }
-      });
+      try {
+        sec.execute(new TxRunnable() {
+          @Override
+          public void run(DatasetContext context) throws Exception {
+            FileSet checkpointFileSet = context.getDataset(DataStreamsApp.CHECKPOINT_FILESET);
+            checkpointBaseRef.set(checkpointFileSet.getBaseLocation());
+          }
+        });
+      } catch (TransactionFailureException e) {
+        throw TransactionsUtility.propagate(e, Exception.class);
```

Comment:
But TransactionsUtility.propagate() will propagate the cause inside the  TransactionFailureException, if there is one present. Isn't that the expectation also here ?